src/desktop/apps/rss/test/routes.test.js

```diff
@@ -0,0 +1,124 @@
+import Articles from '../../../collections/articles.coffee'
+import Backbone from 'backbone'
+import sinon from 'sinon'
+const request = require('rewire')('superagent')
+const routes = require('rewire')('../routes.js')
+
+describe('Routes', () => {
+  let req = {}
+  let res = {}
+  const published_at = new Date().toISOString()
+  let results = [
+    { id: 1, published_at },
+    { id: 2, published_at },
+    { id: 3, published_at },
+  ]
+
+  beforeEach(() => {
+    Backbone.sync = sinon.stub()
+    request.get = sinon.stub().returns({
+      end: cb => {
+        cb(null, {
+          body: {
+            html: '<blockquote class="twitter-tweet">',
+          },
+        })
+      },
+    })
+
+    routes.__set__('sd', { ARTSY_EDITORIAL_CHANNEL: 'foo' })
+    routes.__set__('request', request)
+
+    req = {}
+    res = {
+      render: sinon.stub(),
+      set: sinon.stub(),
+    }
+  })
+
+  describe('#partnerUpdates', () => {
+    it('renders articles', () => {
+      routes.partnerUpdates(req, res)
+      Backbone.sync.args[0][2].success({
+        total: 16088,
+        count: 2,
+        results,
+      })
+      res.render.args[0][0].should.containEql('partner_updates')
+      res.render.args[0][1].articles.length.should.eql(3)
+    })
+  })
+
+  describe('#news', () => {
+    it('renders the rss feed if #findArticlesWithEmbeds rejects', async () => {
+      request.get = sinon.stub().returns({
+        end: cb => {
+          cb(new Error())
+        },
+      })
+      routes.__set__('request', request)
+      routes.news(req, res)
+      await Backbone.sync.args[0][2].success({
+        total: 16088,
+        count: 3,
+        results,
+      })
+      setTimeout(() => {
+        res.render.args[0][0].should.containEql('news')
+        res.render.args[0][1].articles.length.should.eql(3)
+      }, 100)
```

Comment:
Is a setTimeout necessary here?

Reply:
Ah yeah, I remember getting around both ways somehow but I can't seem to find it. Lets merge this for now and I'll send you over the example later.